sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1848,11 +1847,7 @@ void ExecCGCommand::emitInstrumentationData() {
         Program = detail::getSyclObjImpl(SyclKernel)
                       ->getDeviceImage()
                       ->get_program_ref();
-      } else if (nullptr != KernelCG->MSyclKernel) {
-        auto SyclProg = detail::getSyclObjImpl(
-            KernelCG->MSyclKernel->get_info<info::kernel::program>());
```

Comment:
Is it safe to remove this and the other changes in this file? Alternatively you could add a new member function to `kernel_impl` that returns `MProgramImpl`, for example `getProgramImpl()`. Then change this to:

```c++
auto SyclProg = KernelCG->MSyclKernel->getProgramImpl();
```

Reply:
Thanks Steffen [75fcfcd](https://github.com/intel/llvm/pull/6666/commits/75fcfcd94561e53e91a7b4b9a0ab95e97aa3aaf7)

Reply:
Great, thank you! Even looks better than before, if you ask me. 😄